### PR TITLE
Improved code, added some tests

### PR DESCRIPTION
**Pull request by anthoskountouris**

### CODE REVIEW

File: src/main/scala/Character.scala [hunk 1]
```diff
@@ -1,3 +1,4 @@
 // Creation of character class
-class Character(val name:String, val isMale:Boolean, val hasHair:Boolean, val wearsGlasses: Boolean, val wearsHut:Boolean) {
+case class Character(name:String, isMale:Boolean, hasHair:Boolean, wearsGlasses: Boolean, wearsHut:Boolean) {
+
 }
```
Comment (GauravJai) [inline src/main/scala/Character.scala]: Brackets are not required with case classes, those can be removed.

File: src/main/scala/Main.scala [hunk 2]
```diff
@@ -1,58 +1,51 @@
...
 
-    // Player 1 list of characters (board)
-    var listOfCharacters1:List[Character] = List(character1,
-      character2,character3,character4,character5,character6,character7,
-      character8,character9)
-
-    // Player 2 list of characters (board)
-    var listOfCharacters2:List[Character] = List(character1,
+    // List of characters (board)
+    var listOfCharacters:List[Character] = List(character1,
```
Comment (GauravJai) [inline src/main/scala/Main.scala]:
```suggestion
    // Consider using more descriptive variable names and avoid duplication of code wherever possible
    val characters: List[Character] = List(
     Character("Tom", true, true, true, false),
     Character("John", true, true, false, true),
     Character("Anthos", true, false, true, true),
     Character("Kokos", true, true, true, true),
     Character("Will", true, false, false, false),
     Character("April", false, true, true, false),
     Character("Maria", false, true, false, true),
     Character("Andrea", false, true, false, false),
     Character("Connie", false, true, true, true)
  )
```

File: src/main/scala/Main.scala [hunk 3]
```diff
@@ -160,75 +113,32 @@ object Main {
...
-  private def removeQuestionFromQuestionsOfPlayer(questionPicked:String, mapOfQuestions:Map[String, Character => Boolean]):Map[String, Character => Boolean]={
+  def removeQuestionFromQuestionsOfPlayer(questionPicked:String, mapOfQuestions:Map[String, Character => Boolean]):Map[String, Character => Boolean]={
     val updatedMapOfQuestions = mapOfQuestions.removed(questionPicked)
     updatedMapOfQuestions
   }
 
   // Functions that removes the characters with the specified criterion
-  private def removeCharacter(yesOrNo:String, listOfCharacters:List[Character], mapOfQuestions:Map[String, Character => Boolean], randomQuestion:String):List[Character] = {
+  def removeCharacter(yesOrNo:String, listOfCharacters:List[Character], mapOfQuestions:Map[String, Character => Boolean], randomQuestion:String):List[Character] = {
   val trueOrFalse:Boolean = if (yesOrNo=="yes"){
```
Comment (GauravJai) [inline src/main/scala/Main.scala]: // Simplify condition
```suggestion
  val trueOrFalse: Boolean = yesOrNo == "yes"
```

**Review by GauravJai — CHANGES_REQUESTED** [review]
### Suggestion for using small loose components:
instead of writing a game for two players can be just one player for it, and can break the whole code in small components, that could be as below 

* Data Preparation: 
	* a component for preparing the list of characters that could be reused for test class as well. i.e. case class
* Select a random character at the start of game:
	* Write a function to randomly select a character to represent the start of the game.​
* Input/Output Handling:
	* Methods for user input and printing characters. 
* Game Logic:
	* A Methods for picking random characters and questions, removing characters, and determining game outcomes.
* Unit Tests:
	* Separate unit tests for each function to ensure correctness.
* Error Handling:
	* Handling invalid user inputs and other potential errors gracefully.

**Comment by anthoskountouris** [comment]
I removed duplicate code, changed it from two player to just one, created a function that generates the desired data to play the game, created a function which asks user for a valid input, and put the main logic in a function which uses recursion instead of a while loop.

**Review by GauravJai — APPROVED** [review]

**Pull request merged**